### ISSUE TITLE
[BUG] Firefox DevTools does not decode Playwright's HAR response bodies of text resources

### ISSUE THREAD

**Issue by DetachHead** [microsoft/playwright#11532]
**Context:**
## System:
 - OS: Windows 10 10.0.18363
 - Memory: 2.46 GB / 15.78 GB
## Binaries:
 - Node: 16.9.1 - C:\Program Files\nodejs\node.EXE
 - npm: 7.24.1 - C:\Program Files\nodejs\npm.CMD
## Languages:
 - Bash: 5.0.17 - C:\WINDOWS\system32\bash.EXE
## npmPackages:
 - playwright: ^1.18.0 => 1.18.0

**Code Snippet**

```javascript
const context = await browser.newContext({ recordHar: { path: 'asdf.har' } })
const page = await context.newPage()
await page.goto('https://google.com/')
await context.close()
```

**Describe the bug**

har file exported by playwright:
![image](https://user-images.githubusercontent.com/57028336/150447483-ecb30a41-4fd4-4544-97ee-4eb69777df9e.png)

har file exported manually:
![image](https://user-images.githubusercontent.com/57028336/150447712-c2d1e7f0-eed0-48df-8be0-2e0c0876ed3e.png)

**Comment by mxschmitt** [comment]
This works when you load the HAR file into Chromium and does not work in Firefox. Seems like Firefox does not decode the text resources, even tho we say its base64. The [spec](http://www.softwareishard.com/blog/har-12-spec/) does not give any advice when to use base64 and when not, so we use it all the time.

**Comment by DetachHead** [comment]
the reason i used firefox for those screenshots was because in chrome the base64 responses were not being displayed at all, which i assume is somehow related.

when exported from playwright:
![image](https://user-images.githubusercontent.com/57028336/150702360-ffed1aa0-cab2-4fc0-83e9-dc767d9679a6.png)

when exported manually:
![image](https://user-images.githubusercontent.com/57028336/150702408-039a2c77-9dda-4309-b782-b772b943a469.png)

but now that i try this on the google.com example i can't seem to reproduce it, only on the application i'm testing. it only seems to happen with sap batch requests

**Comment by moeinsa** [comment]
good

**Comment by alone-breecher** [comment]
![SVG_XSS](https://user-images.githubusercontent.com/76893049/176115009-e53ae94d-c80a-4e06-9424-a30644bbedfe.svg)

**Comment by alone-breecher** [comment]
![SVG_XSS2](http://0.0.0.0:2006/SVG_XSS.svg)

**Comment by mxschmitt** [comment]
I was able to verify that this is fixed in v1.23. Please file new issues for further questions or bug reports.
